common/lib/dependabot/dependency.rb

```diff
@@ -110,6 +110,67 @@ def display_name
       display_name_builder.call(name)
     end
 
+    def humanized_previous_version
+      # If we don't have a previous version, we *may* still be able to figure
+      # one out if a ref was provided and has been changed (in which case the
+      # previous ref was essentially the version).
+      if previous_version.nil?
+        return ref_changed? ? previous_ref : nil
+      end
+
+      if previous_version.match?(/^[0-9a-f]{40}$/)
+        return previous_ref if ref_changed? && previous_ref
+
+        "`#{previous_version[0..6]}`"
+      elsif version == previous_version &&
+            package_manager == "docker"
```

Comment:
I'm worried it's going to get pretty messy if we start to include package manager specific logic in this class. Is there another way to lookup or inject this logic?

Also, just a heads up that this class is also used in our backend service and we expect to be able to serialize it over the apis. I don't see anything that would break that currently but just something to be aware of!

Reply:
> I'm worried it's going to get pretty messy if we start to include package manager specific logic in this class. Is there another way to lookup or inject this logic?

Another way would be to implement something like [display_name](https://github.com/dependabot/dependabot-core/blob/97722615459c6a08b83aa58226523aedbbe88f18/gradle/lib/dependabot/gradle.rb#L20-L27). I don't love that approach though.

Reply:
Ah, got it, didn't notice this check already existed in the common gem before. In that case I think it's OK to leave it in for now.

If we were to improve it we'll probably want to avoid a subclass as that will make the serialization more complicated. The display_name approach wouldn't have that same concern and would let us use code from the docker gem.